### DISCUSSION TITLE
Allow step up (% per sec) and step down (% per sec) to be variable

### DISCUSSION THREAD

**Discussion by bluespresso** [Rem0o/FanControl.Releases#867] [Ideas]
Allow step up (% per sec) and step down (% per sec) to be variable…

dependent on one parameter:
[the absolute difference between current fan power and the fan power requested by fan curve]
ss a mathematical function or as graph.

Currently you can only set step up and step down to fixed values.
For a near steady state condition, e.g. GPU temperature only changes like between 65°C and 66°C I want step up / step down to be as low as possible like 0,1% / sec because the difference in fan power between 65°C and 66°C is quite high, 9,1%. I can hear it when it changes quickly between these two values.

Unfortunately, 0,1%/sec is not feasible on computer start because it takes quite a while until it spins down and when you start a game because it takes too long to spin up.

A variable step up / step down would make sort of a rubber band behavior where the fan power would not jitter as much just because of 1°C change but at the same time would allow quick adjustment if the conditions change rapidly.

**Comment by bluespresso** [comment]
Further explanation
My GPU temperature sensor only reports whole numbers.
On my main slope of my fan curve for GPU I have a slope of 8% / °C.

![01](https://user-images.githubusercontent.com/58310550/173192331-52d266de-3b6e-4e1e-92a3-5fa9b17a9863.jpg)

When the temperature jumps up or down 1 °C I want step up / step down to be just 0,1%/sec. When the jump happens the fan power would momentarily lag behind by 8%.
If the fan power lag increases due to sudden GPU power drop or rise the step up / step down shall increase proportionally.
I would make it up in a graph similar to this.

![02](https://user-images.githubusercontent.com/58310550/173192352-580c0df0-f69f-4a41-b38b-ebb5339a4038.jpg)

**Reply by Rem0o** [reply]
Keep in mind there is only 256 different steps for the fan to take. The % is a user interface thing, behind the scene, its a 8 bit value.

**Reply by bluespresso** [reply]
Ok, the 8 bit thing can be handled by conversion to percent and back.
What about the main idea?
Variable adjustment speed depending on fan lag?
